src/main/java/net/minecraftforge/fluids/FluidStack.java

```diff
@@ -84,17 +86,46 @@ public static FluidStack loadFluidStackFromNBT(NBTTagCompound nbt)
         {
             return null;
         }
-        if (!nbt.hasKey("FluidName", Constants.NBT.TAG_STRING))
+
+        FluidStack stack;
+        if (nbt.hasKey("Fluid", Constants.NBT.TAG_STRING))
         {
-            return null;
+            ResourceLocation fluid = new ResourceLocation(nbt.getString("Fluid"));
+            if (!ForgeRegistries.FLUIDS.containsKey(fluid))
+            {
+                return null;
+            }
+            else
+            {
+                stack = new FluidStack(ForgeRegistries.FLUIDS.getValue(fluid), nbt.getInteger("Amount"));
+            }
         }
-
-        String fluidName = nbt.getString("FluidName");
-        if (FluidRegistry.getFluid(fluidName) == null)
+        //TODO Drop legacy support for 'FluidNane' in 1.14 or 1.15, as it will be outdated 
+        else if (nbt.hasKey("FluidName", Constants.NBT.TAG_STRING))
+        {
+            String fluidName = nbt.getString("FluidName").toLowerCase();
+            Fluid fluid = null;
+            for (Fluid f : ForgeRegistries.FLUIDS.getValues())
+            {
+                if (f.getRegistryName().getResourcePath().equals(fluidName));
+                {
```

Comment:
Doesn't this handle the transition from "old" registry fluids to "new" dictionary fluids? (If not, ignore me.) I would guess that 99% of fluid dictionary names will be the same as the old registry names (at least for those fluids that will be registered there). I wouldn't say the same thing about the rsl names, because those will then be mod internal names---many mods may take this opportunity to use names that fit their naming schema (as I plan to do with our liquid redstone/glowstone/enderpearls).